Specification.md

```diff
@@ -0,0 +1,15 @@
+### plugins.yaml
+
+The `plugins.yaml` file contains a list of objects, each representing a plug-in.
+
+### Plug-Ins
+
+A Plug-In object consists of the following key-value pairs:
+
+- `name`
+- `url`: A link to the project's github repository.
```

Comment:
Is GitHub a requirement? Or would some other SCM be allowed, like GitLab or BitBucket? (I note the workflows assume GitHub.)

Reply:
At the moment, github is a requirement. Downloading plugins via git would be too much effort, so instead `ESPIM` (kinda the reference implementation atm) only downloads the zip. Other services shouldn't be too hard to support in the future.

A different approach would be to require a `download_link` variable - possibly with a substitute for the version, for example
`https://github.com/EndlessSkyCommunity/world-forge/archive/$version.zip`

Reply:
And would some http-file-download location with a zip-file also be allowed as url? For example `https://example.com/ESPlugins/cargo-controls-plugin-v2.8.1.zip`.
Url stands for "Uniform Resource Locator" and implies some kind of automatic detection of what kind of resource is being located.

Reply:
Not at the moment - status quo is that the library literally does `plugin.url + "/archive/" + plugin.version + ".zip"`.

If we decide to go with the download_link thingy - which im growing increasingly fond of - that would allow any kind of zip.

Reply:
What if we just change the `url` to be the full path to the zip-file (also for GitHub)? Then it works with Github and with any other location where zip-files can be downloaded.

Reply:
It's somewhat redundant with `version`, unless we do substitutes, but i can get behind that. It would require another `website` variable.

Reply:
Yes, having separate variables for the website and for the download-url sounds appropriate.

Reply:
Doing something like the appstream format seems appropriate, where you can specify the artifact link, type, and date.

I'm in favor of making `url` host-agnostic; that we only currently support GitHub repos should not be in the specification: it's an artifact (😉) of the current workflow. We can provide metadata retrieval during `generate` for any real SCM provider through their API, we just need to be motivated to add the support.

Reply:
Doing something like the appstream format seems appropriate, where you can specify the artifact link, type, and date.

I'm in favor of making `url` host-agnostic; that we only currently support GitHub repos should not be in the specification: it's an artifact ( 😉 ) of the current workflow. We can provide metadata retrieval during CI pipeline for any real SCM provider through their API, we just need to be motivated to add the support.